src/Startup.cs

```diff
@@ -31,6 +33,14 @@ public void ConfigureServices(IServiceCollection services)
                     options.SlidingExpiration = true;
 
                 });
+
+            services.AddDbContext<PiggyDbContext>(options =>
+                options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
+            services.AddScoped(provider =>
+            {
+                var db = provider.GetService<PiggyDbContext>();
+                return new UserDatabase(db);
```

Comment:
How does this work? Especially the return is interesting?

Reply:
So I've got this from a tutorial on Youtube, so bear that in mind, but as I understand, if we want to inject an object which does not have an empty constructor ( like UserDatabse, which requires a database context), first we need to get the parameters we will be adding to the constructor, so we take the required parameter/service ( in this place PiggyDbContext) and add it into the constructor. I don't know if I explained this very clearly :D